shopfloor_single_product_transfer/services/single_product_transfer.py

```diff
@@ -0,0 +1,750 @@
+        self, product, location, lot=None, packaging=None
+    ):
+        if product.tracking == "lot":
+            message = self.msg_store.scan_lot_on_product_tracked_by_lot()
+            return self._response_for_select_product(location, message=message)
+
+    def _scan_product__select_move_line_domain(self, product, location, lot=None):
+        domain = [
+            ("location_id", "=", location.id),
+            ("product_id", "=", product.id),
+            ("state", "in", ("assigned", "partially_available")),
+            ("picking_id.user_id", "in", (False, self.env.uid)),
+            ("picking_id.picking_type_id", "in", self.picking_types.ids),
+        ]
+        if lot:
+            lot_domain = [("lot_id", "=", lot.id)]
+            domain = AND([domain, lot_domain])
+        return domain
+
+    def _scan_product__select_move_line(
+        self, product, location, lot=None, packaging=None
+    ):
+        domain = self._scan_product__select_move_line_domain(product, location, lot=lot)
+        move_line = self.env["stock.move.line"].search(domain, limit=1)
+        if move_line:
+            stock = self._actions_for("stock")
+            if self.work.menu.no_prefill_qty:
+                # First, mark move line as picked with qty_done = 0,
+                # so the move wont be split because 0 < qty_done < product_uom_qty
+                stock.mark_move_line_as_picked(move_line, quantity=0)
+                # Then, set the no prefill qty on the move line
+                qty_done = 1
+                if packaging:
+                    qty_done = packaging.qty
+                move_line.qty_done = qty_done
+            else:
+                stock.mark_move_line_as_picked(move_line)
+            return self._response_for_set_quantity(move_line)
+
+    def _scan_product__check_create_move_line(
+        self, product, location, lot=None, packaging=None
+    ):
+        if not self.is_allow_move_create():
+            message = self.msg_store.no_operation_found()
+            return self._response_for_select_product(location, message=message)
+
+    def _scan_product__unreserve_move_line(
+        self, product, location, lot=None, packaging=None
+    ):
+        unreserve = self._actions_for("stock.unreserve")
+        if self.work.menu.allow_unreserve_other_moves:
+            move_lines = self._find_location_move_lines(location, product, lot=lot)
+            response = unreserve.check_unreserve(location, move_lines, product, lot)
+            if response:
+                return response
+            unreserve.unreserve_moves(move_lines, self.picking_types)
+
+    def _scan_product__create_move_line_domain(
+        self, product, location, lot=None, packaging=None
+    ):
+        domain = [
+            ("location_id", "=", location.id),
+            ("product_id", "=", product.id),
+            ("available_quantity", ">", 0),
+            # FIXME: handle the scan of packages later
```

Comment:
We can remove this FIXME, and say that packages should be moved with the `single_pack_transfer` scenario.